### PR TITLE
KOGITO-2601: Move State Control API to `kie-bc-editors`

### PR DESCRIPTION
**Pull request by tiagobento**
See https://issues.redhat.com/browse/KOGITO-2601

Since there are changes on the editors, here's a VSIX file (only DMN and BPMN editors available):
https://drive.google.com/file/d/1A0d6HvxhqPz0ksbwJfsfANG4uwkRSDS3/view?usp=sharing

Related PRs:
https://github.com/kiegroup/kogito-tooling/pull/200
https://github.com/kiegroup/appformer/pull/997

### CODE REVIEW

File: packages/microeditor-envelope/src/EditorFactory.ts [hunk 1]
```diff
@@ -14,20 +14,19 @@
...
-
   /**
    * Returns an Editor based on a LanguageData.
    * Receives a messageBus to be used by the Editor to communicate with the outside of the envelope.
-   * @param languageData The custom LanguageData
-   * @param messageBus The message bus
    */
-  createEditor(languageData: T, messageBus: EnvelopeBusInnerMessageHandler): Promise<AppFormer.Editor>;
+  createEditor(
+    languageData1: LanguageData,
```
Comment (caponetto) [inline packages/microeditor-envelope/src/EditorFactory.ts]: Is there a reason to append this `1` on the name? `languageData1`

File: packages/microeditor-envelope/src/EditorEnvelopeView.tsx [hunk 2]
```diff
@@ -89,7 +87,7 @@ export class EditorEnvelopeView extends React.Component<Props, State> {
       "shift+ctrl+z",
       "Edit | Redo last edit",
       async () => {
-        this.props.stateControlService.redo();
+        this.getEditor()!.redo();
```
Comment (caponetto) [inline packages/microeditor-envelope/src/EditorEnvelopeView.tsx:90]: Is there a chance of editor being `undefined` here? (and also on the change below)
With `!` you imply that the editor won't be `undefined` so I'm just double checking.

File: packages/core-api/src/appformer/Editor.ts [hunk 3]
```diff
@@ -33,4 +33,8 @@ export abstract class Editor extends Component {
   public abstract getContent(): Promise<string>;
 
   public abstract getPreview(): Promise<string | undefined>;
+
+  public abstract undo(): Promise<void>;
+
+  public abstract redo(): Promise<void>;
```
Comment (caponetto) [inline packages/core-api/src/appformer/Editor.ts:39]: :heart_eyes:

**Review by caponetto — COMMENTED** [review]
Awesome! Thanks @tiagobento. I've left only a couple of questions.

File: packages/microeditor-envelope/src/EditorFactory.ts (hunk 1 above)
Comment (tiagobento) [reply packages/microeditor-envelope/src/EditorFactory.ts]: Not at all 😅. Changed.

File: packages/microeditor-envelope/src/EditorEnvelopeView.tsx (hunk 2 above)
Comment (tiagobento) [reply packages/microeditor-envelope/src/EditorEnvelopeView.tsx:90]: Well... theoretically yes, but I didn't wanna clutter the code with a check an all that, so I simply left it like that and we'll have an exception be raised in the case we don't have an editor there for some reason. I'm open to suggestions on how to handle that! Best thing I could think of was a nice `console.error` but I didn't think that justifies 3 more lines of code.

Comment (caponetto) [reply packages/microeditor-envelope/src/EditorEnvelopeView.tsx:90]: +1

**Review by caponetto — APPROVED** [review]

File: packages/kie-bc-editors/src/gwtStateControl/GwtCommandRegistry.ts [hunk 4]
```diff
@@ -14,15 +14,15 @@
...
+import { EnvelopeBusInnerMessageHandler } from "@kogito-tooling/microeditor-envelope";
 import { KogitoEdit } from "@kogito-tooling/core-api";
 
 /**
- * PUBLIC ENVELOPE API
+ * PUBLIC GWT EDITORS API
  * Represents a command registry API to be used on command-based editors.
  */
-export interface KogitoCommandRegistry<T> {
+export interface GwtCommandRegistry<T> {
```
Comment (jomarko) [inline packages/kie-bc-editors/src/gwtStateControl/GwtCommandRegistry.ts]: Shouldn't we rename also corresponding `org.appformer.kogito.bridge.client.stateControl.registry.impl.KogitoCommandRegistry` ?

**Review by jomarko — CHANGES_REQUESTED** [review]
One question to code.

Next it probably break DMN Guided Tour https://issues.redhat.com/browse/KOGITO-764 because I am getting this screen after opening a DMN file.

There is missing error flash message  '... can not read "cookie"...' - was not able to capture it.

Screenshot from 2020-07-08 13-16-45

File: packages/kie-bc-editors/src/gwtStateControl/GwtCommandRegistry.ts (hunk 4 above)
Comment (tiagobento) [reply packages/kie-bc-editors/src/gwtStateControl/GwtCommandRegistry.ts]: I think I prefer reverting it to `KogitoCommandRegistry` if that's okay.

**Comment by tiagobento** [comment]
@jomarko Thanks for testing. I'll upload a new VSIX, since this one was out of date. This bug with the cookie thing was fixed here #202. I'll let you know when a new VSIX is uploaded.

**Comment by tiagobento** [comment]
@jomarko VSIX updated. Please download it here: https://drive.google.com/file/d/1A0d6HvxhqPz0ksbwJfsfANG4uwkRSDS3/view?usp=sharing

**Review by jomarko — APPROVED** [review]
thank you @tiagobento attached vsix works ok.

**Review by paulovmr — APPROVED** [review]

**Pull request merged**